sphinx/writers/latex.py

```diff
@@ -2657,5 +2562,48 @@ def visit_admonition(self, node):
                              (name, admonitionlabels[name]))
         return visit_admonition
 
+    def generate_numfig_format(self, builder):
+        # type: (LaTeXBuilder) -> str
+        warnings.warn('generate_numfig_format() is deprecated.',
+                      RemovedInSphinx40Warning)
+        ret = []  # type: List[str]
+        figure = self.builder.config.numfig_format['figure'].split('%s', 1)
+        if len(figure) == 1:
+            ret.append('\\def\\fnum@figure{%s}\n' %
+                       str(figure[0]).strip().translate(tex_escape_map))
+        else:
+            definition = str(figure[0]).strip().translate(tex_escape_map)
+            ret.append(self.babel_renewcommand('\\figurename', definition))
+            if figure[1]:
+                ret.append('\\makeatletter\n')
+                ret.append('\\def\\fnum@figure{\\figurename\\thefigure%s}\n' %
+                           str(figure[1]).strip().translate(tex_escape_map))
+                ret.append('\\makeatother\n')
+
+        table = self.builder.config.numfig_format['table'].split('%s', 1)
+        if len(table) == 1:
+            ret.append('\\def\\fnum@table{%s}\n' %
+                       str(table[0]).strip().translate(tex_escape_map))
+        else:
+            definition = str(table[0]).strip().translate(tex_escape_map)
+            ret.append(self.babel_renewcommand('\\tablename', definition))
+            if table[1]:
+                ret.append('\\makeatletter\n')
+                ret.append('\\def\\fnum@table{\\tablename\\thetable%s}\n' %
```

Comment:
we will need to fix #5889 here too in this deprecated branch by using for example ``\\thetable %s`` with a space in-between to avoid creating control sequence unknown to LaTeX.